packages/plugin-ext/src/main/browser/plugin-storage.ts

```diff
@@ -43,13 +43,13 @@ export class StorageMainImpl implements StorageMain {
     }
 
     protected toKind(isGlobal: boolean): PluginStorageKind {
-        if (isGlobal) {
+        if (!isGlobal && this.workspaceService.workspace) {
+            return {
+                workspace: this.workspaceService.workspace.resource.toString(),
+                roots: this.workspaceService.tryGetRoots().map(root => root.resource.toString())
+            };
+        } else {
```

Comment:
The `else` is superfluous here. Just return with `undefined` after the `if` block.

Reply:
You are correct. Fixed